### PR TITLE
Fix MM-41977

### PR DESCRIPTION
**Pull request by larkox**
#### Summary
Fix MM-41977

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-41977

### CODE REVIEW

File: server/post.go [hunk 1]
```diff
@@ -54,6 +54,10 @@ func (p *Plugin) postMeeting(creator *model.User, channelID string, topic string
 		return nil, nil, err
 	}
 
+	if !p.API.HasPermissionToChannel(creator.Id, channelID, model.PERMISSION_CREATE_POST) {
+		return nil, nil, errors.New("cannot create post in this channel")
+	}
+
```
Comment (srkgupta) [inline server/post.go:60]: Can we move this check before actually creating the meeting?

**Review by srkgupta — APPROVED** [review]
Thanks @larkox. The changes LGTM.

**Review by hanzei — APPROVED** [review]

**Comment by hanzei** [comment]
@srkgupta Are you the one testing this change or should @DHaussermann do that?

**Comment by srkgupta** [comment]
I haven't had a chance to test the final changes. I will let @DHaussermann do the same and confirm if nothing is broken. Thanks.

**Review by catalintomai — APPROVED** [review]

**Review by DHaussermann — APPROVED** [review]
Tested and passed
- Regression tested the plugin functionality.
- Created meetings in all channel types to ensure this still functions as expected.  
- Tested meeting creation with both member and sysadmin role

LGTM!

**Pull request merged**